src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py

```diff
@@ -1080,3 +1080,37 @@ def test_aggregation_node_failed(self, pf):
         run_dict = run._to_dict()
         assert "error" in run_dict
         assert run_dict["error"] == exception
+
+    def test_get_details_against_partial_completed_run(self, pf: PFClient) -> None:
+        flow_mod2 = f"{FLOWS_DIR}/mod-n/two"
+        flow_mod3 = f"{FLOWS_DIR}/mod-n/three"
+        data_path = f"{DATAS_DIR}/numbers.jsonl"
+        # batch run against data
+        run1 = pf.run(
+            flow=flow_mod2,
+            data=data_path,
+            column_mapping={"number": "${data.value}"},
+        )
+        pf.runs.stream(run1)
+        details1 = pf.get_details(run1)
+        assert len(details1) == 20
+        assert len(details1.loc[details1["outputs.output"] != "(Failed)"]) == 10
+        # assert to ensure inputs and outputs are aligned
+        for _, row in details1.iterrows():
+            if str(row["outputs.output"]) != "(Failed)":
+                assert int(row["inputs.number"]) == int(row["outputs.output"])
+
+        # batch run against previous run
+        run2 = pf.run(
+            flow=flow_mod3,
+            run=run1,
+            column_mapping={"number": "${run.outputs.output}"},
+        )
+        pf.runs.stream(run2)
+        details2 = pf.get_details(run2)
+        assert len(details2) == 10
```

Comment:
Seems the length here becomes 10 which is the valid output length of `run1`. I have a little concern here, since your run2 is not a standard evaluation flow. Let's say your data has another key `groundtruth`, and this run2's column mapping uses both data's ground truth and run1's output, now we know that data has 20 lines, run1 has 20 lines but only 10 is valid, then what's the length of run2's output?

Reply:
I just modified the second flow and submitted a [run](https://ml.azure.com/prompts/flow/bulkrun/run/5964e473-7d2b-4f9d-a95d-ac7e09b6efcb/details?wsid=/subscriptions/96aede12-2f73-41cb-b983-6d11a904839b/resourceGroups/promptflow/providers/Microsoft.MachineLearningServices/workspaces/promptflow-eastus2euap&tid=72f988bf-86f1-41af-91ab-2d7cd011db47) like you described, there are still 10 lines - which I guess executor did some magic for this.

![image](https://github.com/microsoft/promptflow/assets/38847871/3578756f-28c2-4f55-b312-3431742c4913)

Reply:
I just take a look on exectuor/batch enginer implementation: in `_merge_input_dicts_by_line`, they will align the data according to the line number, so that the output length should be aligned with the base run (`run1` for your case).

Reply:
Good seems we don't have execution problem here. The only left question is if we should make run2's output length to 10 instead of 20. 

I personally feel it's a bit weird but not a big problem. Maybe ask for @wangchao1230 's opinions.

Reply:
offline synced with Clement, maybe we can raise this to executor internally.